### PR TITLE
Python SDK: Support seek from end

### LINKED ISSUE #7146: Python SDK Wrapper: Support SEEK_END
The `ObjectReader` class currently raises an `io.UnsupportedOperation` when attempting to seek from the end of an object. 

Unfortunately, this breaks parsing Parquet files (tried with both fastparquet and pyarrow using pandas). 

Parquet files store their metadata in the footer, so the common access pattern is almost always to read the last N bytes, followed by random access to columns based on the metadata retrieved from that footer.

### PR DESCRIPTION
**Pull request by N-o-Z**
Closes #7146

## Change Description

Add support to seek from end

### Background

Required to support reading parquet files which reads the file metadata located at eof

### Testing Details

Added integration testsfor seek and for parquet files

### CODE REVIEW

**Review by ozkatz — CHANGES_REQUESTED** [review]
Overall looks great - I'd urge to cache the size of the object because SEEK_END would be called a lot with parquet files, which will many times make multiple calls..

File: clients/python-wrapper/lakefs/object.py [hunk 1]
```diff
@@ -235,13 +235,16 @@ def seek(self, offset: int, whence: int = 0) -> int:
...
         :raise OSError: if calculated new position is negative
+        :raise io.UnsupportedOperation: If whence value is unsupported
         """
         if whence == os.SEEK_SET:
             pos = offset
         elif whence == os.SEEK_CUR:
-            pos = offset - whence
+            pos = self._pos + offset
+        elif whence == os.SEEK_END:
+            size = self._obj.stat().size_bytes  # Seek end requires us to know the size of the file
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply clients/python-wrapper/lakefs/object.py:246]: We are caching the obj stats(). The stats are actually read from lakeFS only on the first call

File: clients/python-wrapper/requirements.txt [hunk 2]
```diff
@@ -11,4 +11,5 @@ sphinx-autodoc-typehints~=1.25.2
 myst-parser~=2.0.0
 pytest~=7.4.3
 pytest-datafiles~=3.0.0
+pandas~=2.1.4
```
[the remark this answers is not in the document]
Comment (N-o-Z) [reply clients/python-wrapper/requirements.txt:14]: Nope :)

**Review by ozkatz — APPROVED** [review]
THANK YOU

File: clients/python-wrapper/lakefs/object.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ozkatz) [reply clients/python-wrapper/lakefs/object.py:246]: ahh great! I approve this PR

**Pull request merged**